timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -952,7 +952,8 @@ void declare_summed_potential(py::module &m) {
                 return new timemachine::SummedPotential(potentials, params_sizes);
             }),
             py::arg("potentials"),
-            py::arg("params_sizes"));
+            py::arg("params_sizes"),
+            py::keep_alive<1, 2>());
```

Comment:
This to avoid the potentials going out of scope and being cleaned up?

Reply:
Yes, it avoids the need to use the hack I had before introducing a wrapper class whose only purpose was to keep a reference to the potentials.

Reply:
When does it die?

Reply:
Should we fix this more generally? By fixing it only for the summed potential it means the usage of potentials have to be differently based on the user's knowledge of the wrapping layer

Reply:
> When does it die?
@proteneer Without `keep_alive` here, `test_construct_differentiable_interface_fast` fails with a [segfault](https://gist.github.com/mcwitt/f35f2fa2695f4ea1648f60758d4c13ba). I think the cleanup is happening when the [SummedPotential.unbound_impl](https://github.com/proteneer/timemachine/blob/eefac40e06025a49aa1a0ec18b799948aaee85d7/timemachine/lib/potentials.py#L45) returns, since at that point the only Python reference to the C++ `Potential` objects (`impls`) goes out of scope.

Storing a reference to the potentials in the `SummedPotential` object is a partial fix, but doesn't help in cases like [this](https://github.com/proteneer/timemachine/blob/eefac40e06025a49aa1a0ec18b799948aaee85d7/fe/functional.py#L63) where we keep only the implementation (not the full `SummedPotential` wrapper) in scope.

> Should we fix this more generally?
@badisa As far as I understand this particular issue is unique to `SummedPotential` because its implementation stores references to objects allocated on the Python side (which I don't think other existing potentials do?). But @maxentile [mentioned](https://github.com/proteneer/timemachine/pull/537#discussion_r756393280) that we sometimes have this issue elsewhere in the code. Would `keep_alive` be useful in the referenced case?

Reply:
@mcwitt I see what you mean, because we construct the unbound potentials in the function. Thanks